grpc/federation/cel/strings.go

```diff
@@ -0,0 +1,405 @@
+					before, found := strings.CutSuffix(str, suffix)
+					if found {
+						return types.String(before)
+					}
+					return types.String(str)
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("equalFold"),
+			OverloadFunc(createStringsID("equalFold_string_string_bool"), []*cel.Type{cel.StringType, cel.StringType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Bool(strings.EqualFold(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("fields"),
+			OverloadFunc(createStringsID("fields_string_strings"), []*cel.Type{cel.StringType}, cel.ListType(cel.StringType),
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					adapter := types.DefaultTypeAdapter
+					return types.NewStringList(adapter, strings.Fields(args[0].(types.String).Value().(string)))
+				},
+			),
+		),
+		// func FieldsFunc(s string, f func(rune) bool) []string
+		BindFunction(
+			createStringsName("hasPrefix"),
+			OverloadFunc(createStringsID("hasPrefix_string_string_bool"), []*cel.Type{cel.StringType, cel.StringType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Bool(strings.HasPrefix(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("hasSuffix"),
+			OverloadFunc(createStringsID("hasSuffix_string_string_bool"), []*cel.Type{cel.StringType, cel.StringType}, cel.BoolType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Bool(strings.HasSuffix(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("index"),
+			OverloadFunc(createStringsID("index_string_string_int"), []*cel.Type{cel.StringType, cel.StringType}, cel.IntType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Int(strings.Index(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("indexAny"),
+			OverloadFunc(createStringsID("indexAny_string_string_int"), []*cel.Type{cel.StringType, cel.StringType}, cel.IntType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Int(strings.IndexAny(args[0].(types.String).Value().(string), args[1].(types.String).Value().(string)))
+				},
+			),
+		),
+		BindFunction(
+			createStringsName("indexByte"),
+			OverloadFunc(createStringsID("indexByte_string_byte_int"), []*cel.Type{cel.StringType, cel.BytesType}, cel.IntType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					return types.Int(strings.IndexByte(args[0].(types.String).Value().(string), args[1].(types.Bytes).Value().(byte)))
+				},
+			),
+			OverloadFunc(createStringsID("indexByte_string_string_int"), []*cel.Type{cel.StringType, cel.StringType}, cel.IntType,
```

Comment:
Would you explain why we need this `OverloadFunc` even if we already have the index function? 🙏

Reply:
I referred to the implementation linked below. While testing, I encountered an issue when trying to pass a Go `byte` type as an argument. It seems that trying to retrieve it using `cel.bytes` didn't work as expected. By defining the argument as `cel.string` instead, I was able to successfully pass and retrieve the value.

https://github.com/mercari/grpc-federation/blob/b948f38603ce28df91b0b67cd3f0af65b54f1ec4/grpc/federation/cel/time.go#L582-L612

Reply:
Makes sense!